homeassistant/components/light/demo.py

```diff
@@ -53,61 +56,66 @@ def __init__(
         self._effect = effect
 
     @property
-    def should_poll(self):
+    def should_poll(self) -> bool:
         """No polling needed for a demo light."""
         return False
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the light if any."""
         return self._name
 
     @property
-    def brightness(self):
+    def available(self) -> bool:
```

Comment:
I consider it an advanced feature. Marking an entity as unavailable will still trigger a state change and be passed to the event bus.